### PR TITLE
Convert provisioning guide sections into modules

### PR DESCRIPTION
**Pull request by tahliar**
Here are some miscellaneous parts of the guide. A lot of it is just a collection of modules, but now they could be used in other user stories in the future.

### CODE REVIEW

File: guides/common/modules/proc_adding-installation-media.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
+[id="adding-installation-media_{context}"]
+= Adding Installation Media to {Project}
+
+Installation media are sources of packages that {ProjectServer} uses to install a base operating system on a machine from an external repository.
+ifeval::["{build}" == "foreman"]
+When you install the Katello plugin, you can download packages from a Pulp mirror. In this case, installation media are ignored.
```
Comment (spetrosi) [inline guides/common/modules/proc_adding-installation-media.adoc:6]:
```suggestion
When you install the Katello plug-in, you can download packages from a Pulp mirror. In this case, installation media are ignored.
```

File: guides/doc-Provisioning_Guide/topics/Bare_Metal.adoc [hunk 2]
```diff
@@ -183,7 +183,7 @@ To create a host with PXE-less provisioning, complete the following steps:
 . Click the *Operating System* tab, and verify that all fields contain values. Confirm each aspect of the operating system.
 . Click *Resolve* in *Provisioning template* to check the new host can identify the right provisioning templates to use.
 +
-For more information about associating provisioning templates, see xref:Configuring_Provisioning_Resources-Creating_Provisioning_Templates[].
+For more information about associating provisioning templates, see xref:provisioning-templates_provisioning[].
 +
 ifeval::["{build}" == "satellite"]
 . If you use the Katello plugin, click the *Parameters* tab, and ensure that a parameter exists that provides an activation key. If not, add an activation key.
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/topics/Bare_Metal.adoc:189]:
```suggestion
. If you use the Katello plug-in, click the *Parameters* tab, and ensure that a parameter exists that provides an activation key. If not, add an activation key.
```

File: guides/common/modules/con_provisioning-workflow.adoc [hunk 3]
```diff
@@ -1,26 +1,5 @@
...
-
-Virtualization Infrastructure::
-  {Project} connects to virtualization infrastructure services such as {OVirt} and VMware to provision virtual machines from virtual image templates or using the same PXE-based boot methods as bare metal providers.
-
-[[Introduction-Defining_the_Provisioning_Workflow]]
-=== Provisioning Workflow Overview
+[id="provisioning-workflow_{context}"]
+= Provisioning Workflow
 
 The provisioning process follows a basic PXE workflow:
```
Comment (spetrosi) [inline guides/common/modules/con_provisioning-workflow.adoc:5]: On the next line, change Satellite to {Project}:

. You create a host and select a domain and subnet. {Project} requests an available IP address from the DHCP {SmartProxyServer} that is associated with the subnet or from the PostgreSQL database in Satellite.

GitHub won't let me comment on this :-(

File: guides/doc-Provisioning_Guide/master.adoc [hunk 4]
```diff
@@ -10,11 +10,47 @@ endif::[]
...
+== Introduction
 
-include::topics/Provisioning_Contexts.adoc[]
+include::common/modules/con_provisioning-overview.adoc[leveloffset=+2]
+
+include::common/modules/con_provisioning-workflow.adoc[leveloffset=+2]
+
+include::common/modules/con_provisioning-contexts.adoc[leveloffset=+2]
+
+include::common/modules/con_provisioning-templates.adoc[leveloffset=+2]
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/master.adoc:22]: How about moving these two section 1.3. Provisioning Contexts and 1.4. Provisioning Templates to Chapter 2. Configuring Provisioning Resources?
1. By doing this we can move con_provisioning-contexts.adoc closer to  proc_setting-the-provisioning-context.adoc
2. Provisioning Contexts and Provisioning Templates are provisioning resources

Comment (tahliar) [reply guides/doc-Provisioning_Guide/master.adoc:22]: Now that you mention it, I wonder if con_provisioning-contexts.adoc is overkill and could just be a few lines at the start of Setting the Provisioning Context. I'm not sure whether the concept info on its own could potentially be reused in, say, the Planning guide. :shrug:

Comment (spetrosi) [reply guides/doc-Provisioning_Guide/master.adoc:22]: Good point about the reuse. The current Planning guide has sections [3.1. Organizations 3.2. Locations](https://access.redhat.com/documentation/en-us/red_hat_satellite/6.6/html-single/planning_for_red_hat_satellite_6/index#chap-Red_Hat_Satellite-Architecture_Guide-Org_Loc_and_Life_Cycle_Environments) that can be substituted by this concept module in future.

File: guides/common/modules/con_provisioning-templates.adoc [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+.Change logs and history
+
+To view the history of changes applied to a template, navigate to *Hosts* > *Provisioning templates*, select one of the templates, and click *History*. Click *Revert* to override the content with the previous version. You can also revert to an earlier change. Click *Show Diff* to see information about a specific change:
+
+* The *Template Diff* tab displays changes in the body of a provisioning template.
+* The *Details* tab displays changes in the template description.
+* The *History* tab displays the user who made a change to the template and date of the change.
+
+.Types of Provisioning Templates
```
[the remark this answers is not in the document]
Comment (tahliar) [reply guides/common/modules/con_provisioning-templates.adoc:22]: It depends on whether we think each half would ever be reused on its own without the rest (size shouldn't necessarily be a factor). I think probably they could be? You might want to include a short intro to provisioning templates somewhere, or just include the list of types.

File: guides/doc-Provisioning_Guide/master.adoc (hunk 4 above)
Comment (tahliar) [reply guides/doc-Provisioning_Guide/master.adoc:22]: Cool, I'll keep the concept module, then!

**Pull request merged**